flutter_screen_recording_web/lib/flutter_screen_recording_web.dart

```diff
@@ -16,35 +19,107 @@ class WebFlutterScreenRecording extends FlutterScreenRecordingPlatform {
   MediaRecorder mediaRecorder;
   Blob recordedChunks;
   String mimeType;
+  Completer<String> _onStopCompleter;
+  Function(String result) onStop;
+
+  AudioContext _audioContext;
+  MediaStreamAudioDestinationNode _audioDestinationNode;
+
+  //id, source
+  Map<String, MediaStreamAudioSourceNode> _audioSourceNodes;
+
+  WebFlutterScreenRecording();
 
   static registerWith(Registrar registrar) {
     FlutterScreenRecordingPlatform.instance = WebFlutterScreenRecording();
   }
 
   @override
-  Future<bool> startRecordScreen(String name) async {
-    return _record(name, true, false);
+  Future<bool> startRecordScreen(
+    String name, {
+    Function(String) onStop,
+  }) async {
+    return _record(
+      name,
+      true,
+      false,
+      onStop: onStop,
+    );
   }
 
   @override
-  Future<bool> startRecordScreenAndAudio(String name) async {
-    return _record(name, true, true);
+  Future<bool> startRecordScreenAndAudio(
+    String name, {
+    bool recordSystemAudio = true,
+    bool disableUserAudio = false,
+    Function(String) onStop,
+  }) async {
+    return _record(
+      name,
+      true,
+      true,
+      recordSystemAudio: recordSystemAudio,
+      disableUserAudio: disableUserAudio,
+      onStop: onStop,
+    );
   }
 
-  Future<bool> _record(String name, bool recordVideo, bool recordAudio) async {
+  Future<bool> _record(
+    String name,
+    bool recordVideo,
+    bool recordAudio, {
+    bool recordSystemAudio = false,
+    bool disableUserAudio = false,
+    Function(String) onStop,
+  }) async {
     try {
-      var audioStream;
+      this.onStop = onStop;
+      var userMediaStream;
       if (recordAudio) {
-        audioStream = await navigator.getUserMedia({"audio": true});
+        userMediaStream = await navigator.getUserMedia({"audio": true});
       }
-      stream = await navigator
-          .getDisplayMedia({"audio": recordAudio, "video": recordVideo});
+      var displayMediaStream = await navigator.getDisplayMedia({
+        "audio": recordSystemAudio,
+        "video": {
+          'width': {'ideal': 1920},
+          'height': {'ideal': 1080},
```

Comment:
It's resolution of recorded video or the width and height of screen which we are recording ?

Reply:
resolution of recorded video

Reply:
So it's ok 👍